### DISCUSSION TITLE
Model Config 'copy_on_model_validation' deprecated

### DISCUSSION THREAD

**Discussion by detmoo** [pydantic/pydantic#7225] [Pydantic V2]
I am migrating from v1.10 to v2.2.

It looks like the Model Config object in v2.2 no longer includes the '`copy_on_model_validation`' option that was supported in v1.10: it is not included in [ConfigDict API documentation](https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict)). However, this change is not discussed in the Migration guide.

Please can you provide any advice on replicating the '`copy_on_model_validation`' option behaviour from v.1.10 in v2.2? (I'm wondering if it effectively replaced by '`revalidate_instances`'?)

**Comment by Xtreemrus** [comment]
Also encountered this problem, this default behavior is not always expected and I don't know what could be an alternative to the config copy_on_model_validation.

```
from pydantic import BaseModel

class User(BaseModel):
    id: int
    friends: List[int]

    def add_friend(self, friend_id: int):
        self.friends.append(friend_id)

friends = [1, 2, 3]
user = User(id=1, friends=friends)
user.add_friend(4)
```

friends result:  [1, 2, 3]

**Comment by jwarwick-delfi** [comment]
Hi, I am running into this as well, and wanted to add another minimal example for the impact this behavior has on generic types:

```python
from __future__ import annotations
from pydantic import BaseModel
from typing import Any

class GenericBase[T: Any](BaseModel):
	thing: T

class StrImpl(GenericBase[str]):
	pass

class Container(BaseModel):
	model: GenericBase[Any]

c = Container(model=StrImpl(thing="hello, world"))


# >>> c
# Container(model=GenericBase[Any](thing='hello, world'))
# >>> type(c.model)
# <class '__main__.GenericBase[Any]'> 
```
The desired behavior would be:
```python
# >>> c
# Container(model=StrImpl(thing='hello, world'))
# >>> type(c.model)
# <class '__main__.StrImpl'>

```

However I can't find any setting that enables this.
